[2.6] Add NGINX 1.0.4

pennyscissors opened the pull request:
Issue: https://github.com/rancher/rancher/issues/35128
Related PR: https://github.com/rancher/ingress-nginx/pull/67

kinarashah commented on rke/templates/nginx-v0.49.3.go:
```diff
@@ -33,13 +33,18 @@ metadata:
   namespace: ingress-nginx
-{{- if .Options }}
 data:
+{{- if $v := (index .Options "allow-snippet-annotations") }}
```
We should add # Rancher specific change comment here

kinarashah commented on rke/templates/nginx-v0.49.3.go:
```diff
@@ -620,6 +625,8 @@ spec:
                   fieldPath: metadata.namespace
       restartPolicy: OnFailure
       serviceAccountName: ingress-nginx-admission
+      nodeSelector:
```
Is this coming from upstream? nodeSelector already exists above for `ingress-nginx-admission-create` L605, so don't need to add it here.

kinarashah commented on rke/templates/nginx-v0.49.3.go:
```diff
@@ -673,6 +680,8 @@ spec:
                   fieldPath: metadata.namespace
       restartPolicy: OnFailure
       serviceAccountName: ingress-nginx-admission
+      nodeSelector:
```
Same as above, nodeSelector already existing on spec: L658.

kinarashah commented on rke/templates/templates.go:
```diff
@@ -164,8 +165,12 @@ func LoadK8sVersionedTemplates() map[string]map[string]string {
+			">=1.19.15-rancher1-3 <1.20.0-rancher0":   nginxIngressv0493,
+			">=1.20.0-rancher0 <1.20.11-rancher1-2":   nginxIngressV11512,
+			">=1.20.11-rancher1-3 <1.21.0-rancher0":   nginxIngressv0493,
+			">=1.21.0-rancher0 <1.21.5-rancher1-1":    nginxIngressv0481,
```
This should be `<1.21.5-rancher1-2`. Current way would mean 1.21.5-rancher1-1 has no template. 
Also, check if our `go generate` caught and logged this as a warning. If it didn't, could be a bug.

pennyscissors replied:
It did come up, but I missed it. Thanks for pointing it out.

Commit:
Copy previous nginx template

Commit:
Add nginx 1.0.4

Commit:
Copy previous k8s versions

Commit:
Add new k8s versions

Adds the following versions:
1.19.15-rancher1-3
1.20.11-rancher1-3
1.21.5-rancher1-2

Commit:
Go generate

kinarashah commented on rke/k8s_version_info.go:
```diff
@@ -510,6 +522,18 @@ func loadK8sVersionInfo() map[string]v3.K8sVersionInfo {
 			MinRancherVersion: "2.5.6-rc0",
 			MinRKEVersion:     "1.2.0-rc0",
 		},
+		"v1.20.11-rancher1-1": {
```
Note: rancher1-1 and rancher1-2 aren't introduced here, just carrying the version forward from 2.5. Not a behavior change, all versions will load for 2.6 anyway. Adding them now to keep the versions consistent.
